### PR TITLE
 Increase user-friendliness

### PR DESCRIPTION
**Pull request by sylvainschmitt**
To increase user-friendliness (see #62), I have:

### Basic

- [x] Remove checks for unused columns PlotArea
- [x] Remove checks for unused columns VernName, Circ, Xfield, Yfield, Lat, Lon
- [x] Check Plot column only if present
- [x] Check CensusYear column only if present
- [x] Ease basic types inputs with autoconversion tests (with as.integer, as.logical and as.character)
- [x] Ease table types inputs with autoconversion (DT & tibble & data.frame)
- [x] Ease spatial types inputs with autoconversion (raster & terra, sp & sf)
- [x] Add a stop if check/prepare inputs remove all lines
- [x] Add a message if DBH or diameter column exists but not Circ
- [x] Check DBH interval (1cm to 10m)

### Intermediate

- [x] Replace creekdistances argument for a named list with two arguments for two raster inputs
- [x] Add an helper function to create the mask

The new code have passed tests and CRAN checks. I'm open for review if you want to change a few things.

### CODE REVIEW

File: R/getmask.R [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#' Get mask
+#'
```
Comment (VincyaneBadouard) [inline R/getmask.R:2]: tu voudrais bien expliquer ce que tu fais dans un onglet description s'il te plait ? merci !

File: R/inventorycheckformat.R [hunk 2]
```diff
@@ -43,21 +43,24 @@ inventorycheckformat <- function(
...
-    GoodData <- FALSE
-    GeneralStop <- paste (GeneralStop, "Plot variable is not found.")
+    inventory$Plot = "UnknownPlot"
+    warning("The Plot variable is not found, we assume that all data come from one plot.")
   }
   if(!("CensusYear" %in% names(inventory))) {
-    GoodData <- FALSE
-    GeneralStop <- paste (GeneralStop, "CensusYear variable is not found.")
+    inventory$CensusYear = NA_integer_
+    warning("The CensusYear variable is not found, we assume that all data come from one year.")
```
Comment (VincyaneBadouard) [inline R/inventorycheckformat.R:63]: Mettre Censusyear entre guillemets

File: tests/testthat/test-getgeometry.R [hunk 3]
```diff
@@ -21,13 +21,6 @@ test_that("getgeometry ", {
   expect_error(getgeometry (MatrixInventory, var),
                regexp = "The 'inventory' argument of the 'getgeometry ' function must be data.frame")
 
-  # expect_error(getgeometry (inventory, var = "Localisation"),
```
Comment (VincyaneBadouard) [inline tests/testthat/test-getgeometry.R:24]: pourquoi t'as enlevé ça ?

File: tests/testthat/test-loggingsimulation1.R [hunk 4]
```diff
@@ -17,96 +17,90 @@ test_that("loggingsimulation1", {
...
-                                  creekdistances  = CreekDistances, speciescriteria = Matrixspeciescriteria,
+                                  creekverticaldistance = CreekDistances$distvert,
+                                  creekhorizontaldistance = CreekDistances$disthorz,
+                                  speciescriteria = Matrixspeciescriteria,
                                   volumeparameters = MatrixVolParam, scenario = "RIL1",
                                   crowndiameterparameters = MatrixCDparam),
                regexp = "The 'inventory', 'speciescriteria', 'volumeparameters' and 'crowndiameterparameters' arguments
          of the 'loggingsimulation' function must be data.frames")
 
-  # plotmask
```
Comment (VincyaneBadouard) [inline tests/testthat/test-loggingsimulation1.R:39]: tu peux le garder mais mettre un input refusé à plotmask

File: tests/testthat/test-loggingsimulation1.R [hunk 5]
```diff
@@ -17,96 +17,90 @@ test_that("loggingsimulation1", {
...
+                                  creekhorizontaldistance = CreekDistances$disthorz,
+                                  speciescriteria = SpeciesCriteria,
                                   volumeparameters = ForestZoneVolumeParametersTable,
                                   crowndiameterparameters = ParamCrownDiameterAllometry,
                                   scenario = "RIL1",
                                   topography = NULL),
                regexp = "The 'topography' argument of the 'loggingsimulation' function must be a RasterLayer")
 
-  # creekdistances
-  expect_error(loggingsimulation1(Paracou6_2016, plotmask = PlotMask, speciescriteria = SpeciesCriteria,
```
Comment (VincyaneBadouard) [inline tests/testthat/test-loggingsimulation1.R:55]: pareil on peut garder le test mais mettre maintenant un input de classe refusée

File: R/getmask.R (hunk 1 above)
Comment (sylvainschmitt) [reply R/getmask.R:2]: Que veux tu dire par onglet description ? Je veux bien t'expliquer à l'oral si u le souhaites.

File: R/inventorycheckformat.R [hunk 6]
```diff
@@ -119,25 +104,27 @@ inventorycheckformat <- function(
...
     GeneralStop <- paste (GeneralStop, "Yutm variable is not found.")
   }
+
+  if (!GoodData)# inverse value of the object
+    stop(paste ("Input inventory does not comply.", GeneralStop))
+
   #### Variables class check ####
   if (!inherits(inventory$idTree, "integer")) {
-    GoodData <- FALSE
-    GeneralStop <- paste (GeneralStop, "idTree variable should be an integer.")
```
[the remark this answers is not in the document]
Comment (sylvainschmitt) [reply R/inventorycheckformat.R:125]: C'est ton code !

File: R/loggingsimulation1.R [hunk 7]
```diff
@@ -179,26 +187,40 @@ loggingsimulation1 <- function(
...
+      topography <- raster(topography)
+    if(!inherits(topography, "SpatRaster"))
+      stop("The 'topography' argument of the 'loggingsimulation' function must be a RasterLayer")
+  }
 
   # creekdistances
   if(length(creekdistances) < 2 |
      !inherits(creekdistances, "list"))
-    stop("The 'creekdistances' argument of the 'loggingsimulation' function must be a list of 2 elements")
+    stop("The 'creekverticaldistance', and 'creekhorizontaldistance' arguments
```
[the remark this answers is not in the document]
Comment (sylvainschmitt) [reply R/loggingsimulation1.R:222]: Test inutile à supprimer

File: tests/testthat/test-getgeometry.R (hunk 3 above)
Comment (sylvainschmitt) [reply tests/testthat/test-getgeometry.R:24]: Parce que c'était commenté.

File: tests/testthat/test-loggingsimulation1.R (hunk 4 above)
Comment (sylvainschmitt) [reply tests/testthat/test-loggingsimulation1.R:39]: Je gère pas l'ajout de tests.

**Comment by sylvainschmitt** [comment]
I made the last developments required by your review. If the check is validated I think we can consider user-friendliness developments done.

File: R/getmask.R [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+#' @description Build a mask with the intersection of the topography and the spatial inventory.
+#'
+#' @param inventory Input inventory (see the inputs formats and metadata in the
+#'   vignette) (data.frame)
```
Comment (GeraldineDerroire) [inline R/getmask.R:6]: is it the raw or prepared inventory that is required here ? I guess it's the raw and it that true, than nothing to change. If it's the prepared, we have to say it explicitly.

Comment (sylvainschmitt) [reply R/getmask.R:6]: This is the raw one.

File: R/getmask.R [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+#'
+#' @param topography Digital terrain model (DTM) of the inventoried plot (LiDAR,
+#'  1m resolution)
+#'  (RasterLayer **with a crs in UTM**) (See \code{\link{DTMParacou}})
+#'  We advise you to generate your raster with Qgis rather than with the
```
Comment (GeraldineDerroire) [inline R/getmask.R:11]: Why do you suggest to create the raster in QGis ? If you added some code to transform the raster input to the right format, that shouldn't matter no ? And I thought that the package was using the format of the raster package

File: R/getmask.R [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  ## prepare inventory
+  inventory <- inventory %>%
+    select(idTree, Xutm, Yutm) %>%
+    st_as_sf(coords = c("Xutm", "Yutm")) %>% # as sf
+    st_set_crs(st_crs(topography)) # set crs
+  inventory$topography <- extract(topography, inventory)
+  inventory <- inventory %>%
+    filter(!is.na(topography))
+  if(nrow(inventory) < 1)
+    stop("The inventory does not match your topography. Please check your inputs.")
```
Comment (GeraldineDerroire) [inline R/getmask.R:62]: good that you check this

File: R/getmask.R (hunk 9 above)
Comment (sylvainschmitt) [reply R/getmask.R:11]: Sorry this is a copy paste from Vincyane functions, but we could remove it (in that case everywhere). To discuss with @VincyaneBadouard .

File: R/getmask.R (hunk 10 above)
Comment (sylvainschmitt) [reply R/getmask.R:62]: Yes but this is only in `getmask`, we might want to add the check in `logginglab` as the master student recently did the mistake with `CreekDistance` and apparently Margaux alos did it during her internship.

File: R/loggingsummary.R [hunk 11]
```diff
@@ -81,31 +80,31 @@ loggingsummary <- function(x
 
   cat('Initial harvestable volume (m3):\n')
   print(unlist(RsltStats[3]))
-  print(set_units(unlist(RsltStats[3])/PlotArea, m3/ha)) # the harvestable volume (m3) with your initial criteria
+  print(set_units(unlist(RsltStats[3])/HarvestableArea, m3/ha)) # the harvestable volume (m3) with your initial criteria
```
Comment (GeraldineDerroire) [inline R/loggingsummary.R:84]: But I thought that you changed that before Vincyane ? WIS it just in this print that it hasn't been corrected or were there still some problem in the actual calculation of the volume objectif ? See discussion from today with Margaux.

File: R/getmask.R (hunk 10 above)
Comment (GeraldineDerroire) [reply R/getmask.R:62]: indeed, can one of you do it please?

**Comment by sylvainschmitt** [comment]
@VincyaneBadouard , my last version passed all checks. I'm opened to discuss is inclusion.

**Pull request merged**